### PR TITLE
Pin pyOpenSSL major version to 17.* for Python 2.6

### PR DESCRIPTION
**Pull request by SethMichaelLarson**
See: https://github.com/pyca/pyopenssl/blob/74de8a137d435d45c100b74cc971be556166a559/CHANGELOG.rst#1800-2018-05-16

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -63,7 +70,7 @@
       test_suite='test',
       extras_require={
           'secure': [
-              'pyOpenSSL>=0.14',
+              PYOPENSSL_VERSION,
```
Comment (sigmavirus24) [inline setup.py]: This may require a change in setup.cfg too for the wheel

Comment (sigmavirus24) [reply setup.py]: Actually it does: https://github.com/urllib3/urllib3/blob/30e9855307aaa9f9f09651f2bbddeb7350eed666/setup.cfg#L14 Your logic here will need to become more complicated.

Comment (SethMichaelLarson) [reply setup.py]: Dang, reasons we can't have nice things. :P Good catch!

**Comment by SethMichaelLarson** [comment]
Would something to the effect of:
```
pyOpenSSL>=0.14, <18.0.0; python_version=="2.6" and extra == 'secure'
pyOpenSSL>=0.14; python_version=="2.7" and extra == 'secure'
```
be sufficient?

**Comment by sigmavirus24** [comment]
Yes, that should do what you want... I think.

**Comment by SethMichaelLarson** [comment]
Thanks for the review @sigmavirus24!

**Pull request merged**